utils/selectionStateUtils.ts

```diff
@@ -45,13 +45,56 @@ const updateSelectionState = (
   return updatedState;
 };
 
-const selectionStateToString = (selectionState: SelectionState): string => {
+const selectionStateTrueToString = (
+  selectionState: SelectionState | undefined,
+): string => {
+  if (selectionState === undefined) return '';
   const stringRepresentation = Object.entries(selectionState)
     .filter(([, value]) => value)
     .map(([key]) => key)
     .join(', ');
 
-  return stringRepresentation.length > 0 ? `${stringRepresentation},` : '';
+  return stringRepresentation.length > 0 ? `${stringRepresentation}, ` : '';
+};
+
+const stateToValidatedState = (
+  state1: SelectionState | undefined,
+  state2: SelectionState | undefined,
+): SelectionState => {
+  const updatedState: SelectionState = state1 ? { ...state1 } : {};
+  if (!state2) return updatedState;
+
+  const keysToAdd: string[] = [];
+
+  Object.keys(state2).forEach((key) => {
+    if (state2[key] === false || state2[key] === undefined) {
+      delete updatedState[key];
+    } else if (state2[key] && !(key in updatedState)) {
+      keysToAdd.push(key);
```

Comment:
```suggestion
      updatedState[key] = true;
```
j'ai l'impression que ça serait la même chose non?